metrics/ci_integrations/lib/source/github_actions/adapter/github_actions_source_client_adapter.dart

```diff
@@ -136,17 +135,19 @@ class GithubActionsSourceClientAdapter
 
     final run = await _fetchWorkflowRun(buildNumber);
 
-    if (run == null || !_isConclusionValid(run.conclusion)) {
+    if (run == null ||
+        run.status == GithubActionStatus.queued ||
+        !_isConclusionValid(run.conclusion)) {
       return null;
     }
 
     final job = await _fetchJob(run, jobName);
 
-    if (job == null || !_isConclusionValid(job.conclusion)) {
-      return null;
+    if (_isJobValid(job)) {
+      return _mapJobToBuildData(job, run);
     }
```

Comment:
Maybe it would be better to move this check into the `fetchJob` method to avoid duplication?